src/serializer/ResidualHeapVisitor.js

```diff
@@ -194,7 +196,7 @@ export class ResidualHeapVisitor {
       visitedBindings[n] = visitedBinding;
     }
     invariant(visitedBinding.value !== undefined);
-    this.visitValue(visitedBinding.value);
+    visitedBinding.value = this.visitEquivalentValue(visitedBinding.value);
```

Comment:
How did you decide which visitValue calls to place with visitEquivalentValue calls? Shouldn't it be done systematically?